bigtable-hbase/src/main/java/com/google/cloud/bigtable/hbase/BigtableOptionsFactory.java

```diff
@@ -178,66 +227,49 @@ public static BigtableOptions fromConfiguration(Configuration configuration) thr
-    }
+    bigtableOptionsBuilder.setDataHost(getHost(configuration, overrideIpAddress,
+      BIGTABLE_HOST_KEY, BIGTABLE_HOST_DEFAULT, "API Data"));
 
-    String tableAdminHost = configuration.get(
-        BIGTABLE_TABLE_ADMIN_HOST_KEY, BIGTABLE_TABLE_ADMIN_HOST_DEFAULT);
-    if (overrideIpAddress == null) {
-      LOG.debug("Table admin endpoint host %s", tableAdminHost);
-      bigtableOptionsBuilder.setTableAdminHost(InetAddress.getByName(tableAdminHost));
-    } else {
-      LOG.debug("Table admin endpoint host %s. Using override IP address.", tableAdminHost);
-      bigtableOptionsBuilder.setTableAdminHost(
-          InetAddress.getByAddress(tableAdminHost, overrideIpAddress.getAddress()));
-    }
+    bigtableOptionsBuilder.setTableAdminHost(getHost(configuration, overrideIpAddress,
+      BIGTABLE_TABLE_ADMIN_HOST_KEY, BIGTABLE_TABLE_ADMIN_HOST_DEFAULT, "Table Admin"));
 
-    String clusterAdminHost = configuration.get(
-        BIGTABLE_CLUSTER_ADMIN_HOST_KEY, BIGTABLE_CLUSTER_ADMIN_HOST_DEFAULT);
-    if (overrideIpAddress == null) {
-      LOG.debug("Cluster admin endpoint host %s", clusterAdminHost);
-      bigtableOptionsBuilder.setClusterAdminHost(InetAddress.getByName(clusterAdminHost));
-    } else {
-      LOG.debug("Cluster admin endpoint host %s. Using override IP address.", clusterAdminHost);
-      bigtableOptionsBuilder.setClusterAdminHost(
-          InetAddress.getByAddress(clusterAdminHost, overrideIpAddress.getAddress()));
-    }
+    bigtableOptionsBuilder.setClusterAdminHost(getHost(configuration, overrideIpAddress,
+      BIGTABLE_CLUSTER_ADMIN_HOST_KEY, BIGTABLE_CLUSTER_ADMIN_HOST_DEFAULT, "Cluster Admin"));
 
     int port = configuration.getInt(BIGTABLE_PORT_KEY, DEFAULT_BIGTABLE_PORT);
     bigtableOptionsBuilder.setPort(port);
+    return bigtableOptionsBuilder;
+  }
 
-    ThreadFactory threadFactory = new ThreadFactoryBuilder()
-        .setDaemon(true)
-        .setNameFormat(GRPC_EVENTLOOP_GROUP_NAME + "-%d").build();
-    final EventLoopGroup elg = new NioEventLoopGroup(0, threadFactory);
-    bigtableOptionsBuilder.setCustomEventLoopGroup(elg);
+  private static String getValue(final Configuration configuration, String key, String type) {
+    String value = configuration.get(key);
+    Preconditions.checkArgument(
+        !Strings.isNullOrEmpty(value),
+        String.format("%s must be supplied via %s", type, key));
+    LOG.debug("%s %s", type, value);
+    return value;
+  }
 
-    bigtableOptionsBuilder.setChannelOptions(createChannelOptions(configuration));
-    createChannelOptions(configuration).addClientCloseHandler(new Closeable() {
-      @Override
-      public void close() throws IOException {
-        elg.shutdownGracefully();
-      }
-    });
+  private static InetAddress getHost(Configuration configuration, InetAddress overrideIpAddress,
+      String key, String defaultVal, String type) throws UnknownHostException {
+    String hostName = configuration.get(key, defaultVal);
```

Comment:
It's slightly different logic... defaultValue and logging are different for InetAddress host lookup.